tap_github/client.py

```diff
@@ -58,15 +59,15 @@ def get_next_page_token(
         ):
             return None
 
-        resp_json = response.json()
-        if isinstance(resp_json, list):
-            results = resp_json
-        else:
-            results = resp_json.get("items")
+        # Leverage header links returned by the GitHub API.
```

Comment:
@aaronsteers I revamped the next_token logic completely, let me know what you think.

Reply:
I'm pretty confident GitHub sends the next link for all requests but I'll investigate a bit more!

Reply:
https://docs.github.com/en/rest/guides/traversing-with-pagination